File: entities/src/main/java/us/greatapps4you/greatsales/entities/sale/Carrier.java
/*
 * Copyright (c) 2021. GreatApps4you LLC
 * This Software is licenced under the GNU GENERAL PUBLIC LICENSE
 * https://www.gnu.org/licenses/gpl-3.0.txt
 * https://greatapps4you.us
 * CSSML NDSMD VRS + SNMV SMQL IVB
 */

package us.greatapps4you.greatsales.entities.sale;

import us.greatapps4you.greatsales.entities.registration.Address;
import us.greatapps4you.greatsales.entities.registration.Identification;
import java.io.Serializable;
import java.util.UUID;

public class Carrier implements Serializable {

    private Long sequential;
    private UUID uuid;
    private Identification identification;
    private Address address;

    public Carrier() {
    }

    public Carrier(Long sequential, UUID uuid, Identification identification, Address address) {
        this.sequential = sequential;
        this.uuid = uuid;
        this.identification = identification;
        this.address = address;
    }

    public Long getSequential() {
        return sequential;
    }

    public void setSequential(Long sequential) {
        this.sequential = sequential;
    }

    public UUID getUuid() {
        return uuid;
    }

    public void setUuid(UUID uuid) {
        this.uuid = uuid;
    }

    public Identification getIdentification() {
        return identification;
    }

    public void setIdentification(Identification identification) {
        this.identification = identification;
    }

    public Address getAddress() {
        return address;
    }

    public void setAddress(Address address) {
        this.address = address;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Carrier carrier = (Carrier) o;

        if (!identification.equals(carrier.identification)) return false;
        return address.equals(carrier.address);
    }

    @Override
    public int hashCode() {
        int result = identification.hashCode();
        result = 31 * result + address.hashCode();
        return result;
    }

    @Override
    public String toString() {
        return "Carrier{" +
                "identification=" + identification +
                '}';
    }
}

File: desktop/src/main/resources/META-INF/resources/register/js/carrier.js
/*
 * Copyright (c) 2021 GreatApps4you LLC
 * This Software is licenced under the GNU GENERAL PUBLIC LICENSE v3
 * https://www.gnu.org/licenses/gpl-3.0.txt
 * https://greatapps4you.us
 * CSSML NDSMD VRS + SNMV SMQL IVB
 */

const save_url = "http://localhost:8080/carriers/save";
const list_url = "http://localhost:8080/carriers/list";
const remove_url = "http://localhost:8080/carriers/remove/";
const find_url = "http://localhost:8080/carriers/find/";

list();

$(document).ready(function () {

    $("#save").click(function () {

        const identification = {
            name: $("#name").val(),
            tradeName: $("#tradeName").val(),
            taxId: $("#taxId").val()
        };

        const address = {
            street: $("#street").val(),
            number: $("#number").val(),
            complement: $("#complement").val(),
            zip: $("#zip").val(),
            neighborhood: $("#neighborhood").val(),
            city: $("#city").val(),
            countryState: $("#countryState").val(),
            country: $("#country").val(),
            website: $("#website").val(),
            email: $("#email").val(),
            cellPhone: $("#cellPhone").val(),
            phone: $("#phone").val()
        };

        const carrier = JSON.stringify({
            identification: identification,
            address: address
        });

        $.ajax({
            url: save_url,
            type: "POST",
            data: carrier,
            contentType: "application/json",
            dataType: "json"
        }).done(function (savedProduct) {
            clearFields();
            list();
        });
    });
});

function list() {
    $.ajax({
        url: list_url,
        type: "GET",
        dataType: "json"
    }).done(function (carriers) {
        let results_table = "<table>" +
            "<thead>" +
            "<tr>" +
            "<th>RAZÃO SOCIAL</th>" +
            "<th>CNPJ</th>" +
            "<th>CIDADE</th>" +
            "<th></th>" +
            "</tr>" +
            "</thead>" +
            "<tbody>";

        for (let i = 0; i < carriers.length; i++) {
            const uuid = carriers[i].uuid;
            results_table += "<tr>"
                + "<td>" + carriers[i].identification.name + "</td>"
                + "<td>" + carriers[i].identification.taxId + "</td>"
                + "<td>" + carriers[i].address.city + "</td>"
                + "<td><a class='button-link-remove' href='" + remove_url + uuid + "'>X</a></td>"
                + "</tr>";
        }
        results_table += "</tbody>" +
            "</table>";

        if (carriers.length == 0) {
            $("#all_carriers").html("Nenhum Fornecedor Cadastrado");
        } else {
            $("#all_carriers").html(results_table);
        }
    });
}

function clearFields() {
    $(document).ready(function () {
        $("#name").val("");
        $("#tradeName").val("");
        $("#taxId").val("");
        $("#street").val("");
        $("#number").val("");
        $("#complement").val("");
        $("#zip").val("");
        $("#neighborhood").val("");
        $("#city").val("");
        $("#countryState").val("");
        $("#country").val("");
        $("#website").val("");
        $("#email").val("");
        $("#cellPhone").val("");
        $("#phone").val("");
    });
}



